numba/targets/listobj.py

```diff
@@ -460,19 +460,36 @@ def build_list(context, builder, list_type, items):
 @lower_builtin(list, types.IterableType)
 def list_constructor(context, builder, sig, args):
 
-    def list_impl(iterable):
-        res = []
-        res.extend(iterable)
-        return res
+    if context._disable_reflected_list:
+        from numba.typed import List
+        def list_impl(iterable):
+            res = List()
+            res.extend(iterable)
+            return res
+    else:
+        def list_impl(iterable):
+            res = []
+            res.extend(iterable)
+            return res
 
     return context.compile_internal(builder, list_impl, sig, args)
 
 @lower_builtin(list)
 def list_constructor(context, builder, sig, args):
-    list_type = sig.return_type
-    list_len = 0
-    inst = ListInstance.allocate(context, builder, list_type, list_len)
-    return impl_ret_new_ref(context, builder, list_type, inst.value)
+    if context._disable_reflected_list:
```

Comment:
In an ideal world we'd pull out all the lowering functions into a new module and then the two branches in the lowering functions for typed/reflected would get their own local imports of the things they need so as to prevent accidents. However, the scope is seemingly quite limited here and it looks ok.

Reply:
Yeah, this appear to be bolted-on somewhat.